proj/hps_accel/vizier_autotune.ipynb

```diff
@@ -0,0 +1,295 @@
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! pip install -U google-api-python-client\n",
+    "! pip install -U google-cloud\n",
+    "! pip install -U google-cloud-storage\n",
+    "! pip install -U requests\n",
+    "! pip install -U matplotlib\n",
+    "! pip install -U scipy"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Synthesize design\n",
+    "\n",
+    "Let's synthesize design beforehand. After all we want to test only nextpnr."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "# Specify a path for your riscv toolchain installation.\n",
+    "# Might be unncessary if you are running inside a conda environment\n",
+    "riscv_toolchain_path = '../../../riscv64-unknown-elf-gcc-10.1.0-2020.08.2-x86_64-linux-ubuntu14/'\n",
+    "riscv_toolchain_path = os.path.realpath(riscv_toolchain_path)\n",
+    "\n",
+    "rv_bin = os.path.join(riscv_toolchain_path, 'bin')\n",
+    "path = f'{rv_bin}:{os.environ[\"PATH\"]}'\n",
+    "\n",
+    "! source ../../env/conda/bin/activate cfu-common\n",
+    "%env PATH $path\n",
+    "! make synth"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Configure SaaS backend and authorization"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Google Cloud Vizier config\n",
+    "gc_accnt = 'example@email.com'\n",
+    "gc_project_id = 'project_id'\n",
+    "gc_region= 'region'\n",
+    "%env GOOGLE_APPLICATION_CREDENTIALS credentials.json"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create study"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import random\n",
+    "import datetime\n",
+    "from vizier.vapi import *\n",
+    "\n",
+    "random.seed()\n",
+    "\n",
+    "def make_study_name(identity):\n",
+    "    random_id = ''.join(str(random.choice([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])) for i in range(0, 8))\n",
+    "    return f'hps-study{random_id}_{identity}_{datetime.datetime.now().strftime(\"%Y%m%d_%H%M%S\")}'\n",
+    "\n",
+    "study_id = make_study_name('hps')\n",
+    "\n",
+    "print(f'Study will be named \"{study_id}\"')\n",
+    "\n",
+    "study = Study(\n",
+    "    name = study_id,\n",
+    "    max_trials = 3,\n",
+    "    # Add metrics to optimize here\n",
+    "    metrics = {\n",
+    "        'fmax': 'MAXIMIZE',\n",
+    "        'runtime': 'MINIMIZE'\n",
+    "    },\n",
+    "    # Add parameters for the blackbox here.\n",
+    "    # Use 'UNSPECIFIED'/'LENEAR'/'LOG'/'REVERSE_LOG' to define parameter scaling (Vizier)\n",
+    "    parameters = {\n",
+    "        'estimate-delay-mult': ParameterInt(min=10, max=50, scale='LINEAR')\n",
```

Comment:
consider using https://papermill.readthedocs.io/en/latest/usage-parameterize.html pattern to parametize the notebook, so that it can easily be batched using: https://cloud.google.com/vertex-ai/docs/workbench/managed/schedule-managed-notebooks-run